### PR TITLE
Bluetooth: Mesh: Various coverity fixes 

### PR DESCRIPTION
**Pull request by trond-snekvik**
Resolves all open Coverity issues on the Bluetooth Mesh subsystem.

### CODE REVIEW

File: subsys/bluetooth/mesh/heartbeat.c [hunk 1]
```diff
@@ -100,8 +100,10 @@ static int heartbeat_send(const struct bt_mesh_send_cb *cb, void *cb_data)
 		.xmit = bt_mesh_net_transmit_get(),
 	};
 
-	/* Do nothing if heartbeat publication is not enabled */
-	if (pub.dst == BT_MESH_ADDR_UNASSIGNED) {
+	/* Do nothing if heartbeat publication is not enabled or the subnet is
+	 * removed.
+	 */
+	if (!tx.sub || pub.dst == BT_MESH_ADDR_UNASSIGNED) {
 		return 0U;
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/heartbeat.c:107]: I know you didn't touch this, but the `0U` return is odd since the return value is not unsigned, i.e. the `U` should probably be removed (can be a separate patch)

**Review by jhedberg — APPROVED** [review]

File: subsys/bluetooth/mesh/heartbeat.c (hunk 1 above)
Comment (trond-snekvik) [reply subsys/bluetooth/mesh/heartbeat.c:107]: True. I'll add a commit to this pr

**Review by nashif — APPROVED** [review]
